### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by yyzNato**

### CODE REVIEW

File: src/components/album/album-comments/comment-list/comment-list-item/comment-list-item.component.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import template from './comment-list-item.html';
+
+export const CommentListItemComponent = {
+	template,
+	bindings: {
+		comment: '<'
+	},
+	controller: class CommentListItemComponent{}
```
Comment (nalejandroveron) [inline src/components/album/album-comments/comment-list/comment-list-item/comment-list-item.component.js]: make sense to put an empty controller?

File: src/components/album/album-comments/comment-list/comment-list.component.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import template from './comment-list.html';
+
+export const CommentListComponent = {
+	template,
+	bindings: {
+		comments: '<'
+	},
+	controller: class CommentListComponent{}
```
Comment (nalejandroveron) [inline src/components/album/album-comments/comment-list/comment-list.component.js]: make sense to put an empty controller?

File: src/components/album/album-comments/comment-new/comment-new.component.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import template from './comment-new.html';
+
+export const CommentNewComponent = {
+	template,
+	bindings: {
+		addComment: '&'
+	},
+	controller: class CommentNewComponent{
+		constructor(){
+			this.email = '';
```
Comment (nalejandroveron) [inline src/components/album/album-comments/comment-new/comment-new.component.js]: We should not initialize things into the contructor, use the `$onInit` hook

File: src/components/album/album-comments/comment-new/comment-new.component.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+			this.text = '';
+		}
+
+		newComment(){
+			if(this.email && this.text.length){
+				let newComment = {
+				  albumId: '',
+				  email: this.email,
+				  text: this.text,
+				  avatar: '',
```
Comment (nalejandroveron) [inline src/components/album/album-comments/comment-new/comment-new.component.js]: Why you set this property empty?

File: src/components/album/album-comments/comment-new/comment-new.component.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+	controller: class CommentNewComponent{
+		constructor(){
+			this.email = '';
+			this.text = '';
+		}
+
+		newComment(){
+			if(this.email && this.text.length){
+				let newComment = {
+				  albumId: '',
```
Comment (nalejandroveron) [inline src/components/album/album-comments/comment-new/comment-new.component.js]: This property should not be empty

File: src/components/album/album-comments/comment.component.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+export const CommentComponent = {
+	template,
+	bindings: {
+		albumId: '<'
+	},
+	controller: class CommentComponent{
+		constructor(AlbumService){
+			'ngInject';
+			this.albumService = AlbumService;
+			this.comments;
```
Comment (nalejandroveron) [inline src/components/album/album-comments/comment.component.js]: This line literally doesn't do anything

File: src/components/album/album-comments/comment.component.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+		getComments(){
+			this.albumService.getCommentsByAlbumId(this.albumId).then(response => {
+				this.comments = response.data;
+			});
+		}
+
+		addComment({comment}){
+			comment.albumId = this.albumId;
+			this.albumService.postNewComment(comment).then(response => {this.getComments()});
```
Comment (nalejandroveron) [inline src/components/album/album-comments/comment.component.js]: The comment component should not directly push a comment, use properties instead

File: src/components/album/album-list/album-list-item/album-list-item.html [hunk 8]
```diff
@@ -0,0 +1,17 @@
+<div class="album-list-item-wrapper">
+	<div class="album-list-item-img">
+		<img ng-src="{{$ctrl.album.images[0].url}}" alt="cover">	
+	</div>	
+	<div class="album-list-item-info">		
+		<div class="album-list-item-details">
+			<h3>{{ $ctrl.album.artists[0]["name"] }}</h3>
```
Comment (nalejandroveron) [inline src/components/album/album-list/album-list-item/album-list-item.html]: Why we're accessing a property using brackets?

File: src/components/album/album.service.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+export class AlbumService{
+	
+	constructor($q, $http){
+		'ngInject';
+		this.$http = $http;
+	}
+
+	getAlbumsByName(albumName) {
+	return this.$http.get(`/search?q=${albumName}`);
```
Comment (nalejandroveron) [inline src/components/album/album.service.js]: Use config to set the params
https://docs.angularjs.org/api/ng/service/$http

Comment (nalejandroveron) [reply src/components/album/album.service.js]: Keep a track of the identation rules

**Review by nalejandroveron — CHANGES_REQUESTED** [review]

**Comment by nalejandroveron** [comment]
In overall this code still need some work, aside of the comments that i've done still need to achieve some points:

- There should be components that defer actions to their parents, instead of doing it inside those "View components".

- The application should use multiple routes instead of being one, this is a must-have

- There are some services not used correctly, like $http

- Some components doesn't respect a single responsibility, ie they do directly calls to backend

- Some components initialize data in the contructor

- The overall design should be worked more, because is unusable at the point of styling.

File: src/components/album/album-comments/comment-list/comment-list-item/comment-list-item.component.js (hunk 1 above)
Comment (yyzNato) [reply src/components/album/album-comments/comment-list/comment-list-item/comment-list-item.component.js]: I've got it

File: src/components/album/album-comments/comment-new/comment-new.component.js (hunk 3 above)
Comment (yyzNato) [reply src/components/album/album-comments/comment-new/comment-new.component.js]: Cool, I'll keep it in mind

File: src/components/album/album-comments/comment-new/comment-new.component.js (hunk 4 above)
Comment (yyzNato) [reply src/components/album/album-comments/comment-new/comment-new.component.js]: It's like the blueprint of the object, but yes, I will set the property when I have already the value

File: src/components/album/album-comments/comment.component.js (hunk 6 above)
Comment (yyzNato) [reply src/components/album/album-comments/comment.component.js]: My bad

File: src/components/album/album-comments/comment.component.js (hunk 7 above)
Comment (yyzNato) [reply src/components/album/album-comments/comment.component.js]: Sorry, what do you mean by using properties?

File: src/components/album/album-list/album-list-item/album-list-item.html (hunk 8 above)
Comment (yyzNato) [reply src/components/album/album-list/album-list-item/album-list-item.html]: Yeah I know that 'name' is a valid identifier (don't contain spaces or key words), so is not necessary using bracket notation

File: src/components/album/album.service.js (hunk 9 above)
Comment (yyzNato) [reply src/components/album/album.service.js]: Indeed, for long query strings it's much clearer

File: src/components/album/album-comments/comment.component.js (hunk 7 above)
Comment (nalejandroveron) [reply src/components/album/album-comments/comment.component.js]: you can use the properties or bindings, to do an expression binding of the object, using `&` https://docs.angularjs.org/guide/component

Comment (yyzNato) [reply src/components/album/album-comments/comment.component.js]: So you're saying that the main container (album.component) should handle that operation

**Comment by nalejandroveron** [comment]
Yes

On Feb 1, 2017 6:03 PM, "yyzNato" <notifications@github.com> wrote:

*@yyzNato* commented on this pull request.
------------------------------

In src/components/album/album-comments/comment.component.js
<https://github.com/LateralView/spotify-exercise/pull/5>:

> +			this.getComments();
+		}
+
+		toggleNewComment(){
+			this.enableNewComment = !this.enableNewComment;
+		}
+
+		getComments(){
+			this.albumService.getCommentsByAlbumId(this.albumId).then(response => {
+				this.comments = response.data;
+			});
+		}
+
+		addComment({comment}){
+			comment.albumId = this.albumId;
+			this.albumService.postNewComment(comment).then(response =>
{this.getComments()});

So you're saying that the main container (album.component) should handle
that operation

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/LateralView/spotify-exercise/pull/5>, or mute the thread

**Pull request closed without merging**